test/cppsuite/tests/cache_resize.cpp

```diff
@@ -0,0 +1,235 @@
+
+        /* Open a cursor on the tracking table to read it. */
+        scoped_session session = connection_manager::instance().create_session();
+        scoped_cursor cursor = session.open_scoped_cursor(operation_table_name);
+
+        /*
+         * Parse the tracking table. Each operation is tracked and each transaction is made of
+         * multiple operations, hence we expect multiple records for each transaction. We only need
+         * to verify that the cache size was big enough when the transaction was committed, which
+         * means at the last operation.
+         */
+        while ((ret = cursor->next(cursor.get())) == 0) {
+
+            uint64_t tracked_ts, tracked_txn_id;
+            int tracked_op_type;
+            const char *tracked_cache_size;
+
+            testutil_check(cursor->get_key(cursor.get(), &tracked_ts, &tracked_txn_id));
+            testutil_check(cursor->get_value(cursor.get(), &tracked_op_type, &tracked_cache_size));
+
+            logger::log_msg(LOG_TRACE,
+              "Timestamp: " + std::to_string(tracked_ts) +
+                ", transaction id: " + std::to_string(tracked_txn_id) +
+                ", cache size: " + std::to_string(std::stoull(tracked_cache_size)));
+
+            tracking_operation op_type = static_cast<tracking_operation>(tracked_op_type);
+            /* There are only two types of operation tracked. */
+            testutil_assert(
+              op_type == tracking_operation::CUSTOM || op_type == tracking_operation::INSERT);
+
+            /*
+             * There is nothing to do if we are reading a record that indicates a cache size change.
+             */
+            if (op_type == tracking_operation::CUSTOM)
+                continue;
+
+            if (first_record) {
+                first_record = false;
+            } else if (prev_txn_id != tracked_txn_id) {
+                /*
+                 * We have moved to a new transaction, make sure the cache was big enough when the
+                 * previous transaction was committed.
+                 *
+                 * FIXME-WT-9339 - Somehow we have some transactions that go through while the cache
+                 * is very low. Enable the check when this is no longer the case.
+                 * testutil_assert(cache_size > cache_size_500mb);.
+                 */
+            }
+            prev_txn_id = tracked_txn_id;
+            /* Save the last cache size seen by the transaction. */
+            cache_size = std::stoull(tracked_cache_size);
+            ++num_records;
+        }
+        /* All records have been parsed, the last one still needs the be checked. */
+        testutil_assert(ret == WT_NOTFOUND);
+        testutil_assert(num_records > 0);
+        /*
+         * FIXME-WT-9339 - Somehow we have some transactions that go through while the cache is very
+         * low. Enable the check when this is no longer the case. testutil_assert(cache_size >
+         * cache_size_500mb);.
```

Comment:
Place code statement in newline 😄

Reply:
@jiechenbo `s_all` does this though, what should I do to make it look better?

Reply:
Fixed in f66a881a2ade6f05c511789f8a8546dcb0858be9